src/app/dataset-view/additional-components/metadata-component/components/add-polling-source/add-polling-source.component.ts

```diff
@@ -0,0 +1,214 @@
+/* eslint-disable @typescript-eslint/no-unsafe-assignment */
+import {
+    FetchKind,
+    ReadKind,
+    MergeKind,
+    PrepareKind,
+    PreprocessKind,
+} from "./add-polling-source-form.types";
+import { FinalYamlModalComponent } from "../final-yaml-modal/final-yaml-modal.component";
+import { SetPollingSource } from "./../../../../../api/kamu.graphql.interface";
+import { BaseComponent } from "src/app/common/base.component";
+import {
+    ChangeDetectionStrategy,
+    ChangeDetectorRef,
+    Component,
+    OnInit,
+} from "@angular/core";
+import { FormBuilder, FormGroup } from "@angular/forms";
+import { AppDatasetCreateService } from "src/app/dataset-create/dataset-create.service";
+import { TemplatesYamlEventsService } from "src/app/services/templates-yaml-events.service";
+import { ActivatedRoute, ParamMap } from "@angular/router";
+import { requireValue } from "src/app/common/app.helpers";
+import { DatasetInfo } from "src/app/interface/navigation.interface";
+import ProjectLinks from "src/app/project-links";
+import { NgbModal, NgbModalRef } from "@ng-bootstrap/ng-bootstrap";
+import { SetPollingSourceSection } from "src/app/shared/shared.types";
+import {
+    FETCH_STEP_RADIO_CONTROLS,
+    MERGE_STEP_RADIO_CONTROLS,
+    READ_STEP_RADIO_CONTROLS,
+} from "./form-control.source";
+import { FETCH_FORM_DATA } from "./steps/data/fetch-form-data";
+import { READ_FORM_DATA } from "./steps/data/read-form-data";
+import { MERGE_FORM_DATA } from "./steps/data/merge-form-data";
+import { ProcessFormService } from "./process-form.service";
+
+@Component({
+    selector: "app-add-polling-source",
+    templateUrl: "./add-polling-source.component.html",
+    styleUrls: ["./add-polling-source.component.sass"],
+    changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class AddPollingSourceComponent extends BaseComponent implements OnInit {
+    public currentStep: SetPollingSourceSection = SetPollingSourceSection.READ;
+    public steps: typeof SetPollingSourceSection = SetPollingSourceSection;
+    public isAddPrepareStep = false;
+    public isAddPreprocessStep = false;
+    public errorMessage = "";
+    // --------------------------------
+    private readonly defaultPrepareKind = PrepareKind.PIPE;
+    private readonly defaultPreprocessKind = PreprocessKind.SQL;
+    // ---------------------------------
+    public readonly fetchStepRadioData = FETCH_STEP_RADIO_CONTROLS;
+    public readonly fetchFormData = FETCH_FORM_DATA;
+    public readonly fetchDefaultKind = FetchKind.URL;
+    // ---------------------------------
+    public readonly readStepRadioData = READ_STEP_RADIO_CONTROLS;
+    public readonly readFormData = READ_FORM_DATA;
+    public readonly readDefaultKind = ReadKind.CSV;
+    // ---------------------------------
+    public readonly mergeStepRadioData = MERGE_STEP_RADIO_CONTROLS;
+    public readonly mergeFormData = MERGE_FORM_DATA;
+    public readonly mergeDefaultKind = MergeKind.APPEND;
```

Comment:
A general rule - whenever we have readonly data, it should be named UPPER_CASE_WITH_UNDERSCORES. Everywhere. I think I explained why this is important a few times already.

Reply:
Fixed